realm/realm-jni/src/io_realm_internal_table.cpp

```diff
@@ -1551,7 +1551,7 @@ bool check_valid_primary_key_column(JNIEnv* env, Table* table, StringData column
                     int64_t next_val = results.get_int(column_index, i);
                     if (val == next_val) {
                         std::ostringstream error_msg;
-                        error_msg << "Field \"" << table->get_column_name(column_index).data() << "\" cannot be a primary key, ";
```

Comment:
I guess that `std::basic_ostream<C,T>& operator<<(std::basic_ostream<C,T>& out, const StringData& d)` is called, and then each character is added to the `ostringstream` one by one?